File: app/CDoc/c/src/pages/chpass.c
// Copyright 20-Aug-2019 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

#include "pages/chpass.h"
#include "dmc/cgi.h"

// mrq is Map[Js]
char *chpass_process (Map *mrq) {
  CGI_GET_STR(user, mrq)
  CGI_GET_STR(pass, mrq)
  CGI_GET_STR(newPass, mrq)

  return cgi_change_pass(user, pass, newPass);
}

File: app/MMarket/go/server/settingsMod/chpass/chpass.go
// Copyright 05-Apr-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// ChangePass source hub.
package chpass

import (
	"github.com/dedeme/golib/cgi"
	"github.com/dedeme/golib/json"
)

func Process(ac chan func(), ck string, mrq map[string]json.T) string {
  return cgi.ChangePass(
    ck,
    cgi.RqString(mrq, "user"),
    cgi.RqString(mrq, "old"),
    cgi.RqString(mrq, "new"),
  )
}
